quickwit-metastore/migrations/postgresql/3_add-split-publish-timestamp-field.up.sql

```diff
@@ -0,0 +1,20 @@
+ALTER TABLE splits
+    ADD COLUMN publish_timestamp TIMESTAMP DEFAULT NULL;
+
+-- We want to update the split `publish_timestamp` field whenever the split
+-- being is published.
+CREATE OR REPLACE FUNCTION set_split_publish_timestamp_for_split() RETURNS trigger AS $$
+BEGIN
+    IF (TG_OP = 'UPDATE') AND (NEW.split_state = 'Published') THEN
```

Comment:
What happens if the split is already published? Shall we add `AND (OLD.split_state != 'Staged')`?